net/dnsfallback/dnsfallback.go

```diff
@@ -90,7 +90,7 @@ func Lookup(ctx context.Context, host string) ([]netaddr.IP, error) {
 // serverName and serverIP of are, say, "derpN.tailscale.com".
 // queryName is the name being sought (e.g. "controlplane.tailscale.com"), passed as hint.
 func bootstrapDNSMap(ctx context.Context, serverName string, serverIP netaddr.IP, queryName string) (dnsMap, error) {
-	dialer := netns.NewDialer()
+	dialer := netns.NewDialer(log.Printf)
```

Comment:
missing plumbing here.

I wonder if we should have some `logfFromContext(ctx)` func that defaults to log.Printf.

Reply:
Most of the callers of bootstrapDNSMap are near startup code (including logging subsystem startup) and use log.Printf themselves. I could thread it a few levels deeper, but it'd still be log.Printf, so I stopped here for now.

> I wonder if we should have some logfFromContext(ctx) func that defaults to log.Printf.

And then start passing around contexts instead of logf everywhere? That'd be really nice. And it'd make changing our logging system a lot easier, if we ever actually do that. cc @danderson 

Work for another day, though.